packages/tools/src/tools/annotation/VideoRedactionTool.ts

```diff
@@ -0,0 +1,935 @@
+//         worldBoundingBox: {
+//           topLeft: Types.Point3;
+//           topRight: Types.Point3;
+//           bottomLeft: Types.Point3;
+//           bottomRight: Types.Point3;
+//         };
+//       };
+//     };
+//     cachedStats: any;
+//     active: boolean;
+//   };
+// }
+
+class VideoRedactionTool extends AnnotationTool {
+  _throttledCalculateCachedStats: any;
+  editData: {
+    annotation: any;
+    viewportUIDsToRender: string[];
+    handleIndex?: number;
+    movingTextBox: boolean;
+    newAnnotation?: boolean;
+    hasMoved?: boolean;
+  } | null;
+  _configuration: any;
+  isDrawing: boolean;
+  isHandleOutsideImage: boolean;
+
+  constructor(toolConfiguration = {}) {
+    super(toolConfiguration, {
+      supportedInteractionTypes: ['Mouse', 'Touch'],
+      configuration: { shadow: true, preventHandleOutsideImage: false },
+    });
+
+    this._throttledCalculateCachedStats = throttle(
+      this._calculateCachedStats,
+      100,
+      { trailing: true }
+    );
+  }
+
+  // TODO_JAMES -> We are just adding this as a "global redaction for now".
+  // I.e. annotation tied to entire video, rather than a frame or a particular time
+  // Range, which would also be possible.
+  addNewAnnotation = (evt: CustomEvent) => {
+    const eventData = evt.detail;
+    const { currentPoints, element } = eventData;
+    const worldPos = currentPoints.world;
+
+    const enabledElement = getEnabledElement(element);
+    const { viewport, renderingEngine } = enabledElement;
+
+    this.isDrawing = true;
+
+    const annotation = {
+      metadata: {
+        // We probably just want a different type of data here, hacking this
+        // together for now.
+        viewPlaneNormal: <Types.Point3>[0, 0, 1],
+        viewUp: <Types.Point3>[0, 1, 0],
+        FrameOfReferenceUID: viewport.getFrameOfReferenceUID(),
+        referencedImageId: viewport.getFrameOfReferenceUID(),
+        toolName: this.getToolName(),
+      },
+      data: {
+        invalidated: true,
+        handles: {
+          points: [
+            <Types.Point3>[...worldPos],
+            <Types.Point3>[...worldPos],
+            <Types.Point3>[...worldPos],
+            <Types.Point3>[...worldPos],
+          ],
+          textBox: {
+            hasMoved: false,
+            worldPosition: <Types.Point3>[0, 0, 0],
+            worldBoundingBox: {
+              topLeft: <Types.Point3>[0, 0, 0],
+              topRight: <Types.Point3>[0, 0, 0],
+              bottomLeft: <Types.Point3>[0, 0, 0],
+              bottomRight: <Types.Point3>[0, 0, 0],
+            },
+          },
+          activeHandleIndex: null,
+        },
+        cachedStats: {},
+        active: true,
+      },
+    };
+
+    addAnnotation(annotation, element);
+
+    const viewportUIDsToRender = getViewportIdsWithToolToRender(
+      element,
+      this.getToolName(),
+      false // TODO_JAMES: Just trying to bypass the vtk specific stuff as fast as possible.
+    );
```

Comment:
I believe this should be at the ned of the render before setting the this.editData